ddtrace/appsec/_iast/_ast/iastpatch.c

```diff
@@ -398,7 +394,7 @@ static int
 str_in_list(const char* needle, const char** list, size_t count)
 {
     for (size_t i = 0; i < count; i++) {
-        if (strcmp(needle, list[i]) == 0) {
+        if (strncmp(needle, list[i], strlen(list[i])) == 0) {
```

Comment:
Are we sure we want that? this will be slower (additional strlen) and not the same semantic, as it will be true if list[i] is a prefix of needle, even if not equal.

Reply:
Yes, that's my intention. The problem is that we wanted to do `needle.startswith(list[i])`. The benchmarks haven't shown any performance degradation (unlike previous commits in this branch with other tests), so we can be confident that there's no performance impact. Thanks for pointing this out!